yoti-sdk-api/src/main/java/com/yoti/api/client/spi/remote/call/identity/Receipt.java

```diff
@@ -44,28 +45,28 @@ public String getParentRememberMeId() {
         return parentRememberMeId;
     }
 
-    public ApplicationProfile getProfile() {
-        return profile;
+    public OffsetDateTime getTimestamp() {
+        return timestamp;
     }
 
-    public HumanProfile getOtherPartyProfile() {
-        return otherPartyProfile;
+    public Optional<HumanProfile> getProfile() {
+        return userContent.getProfile();
     }
 
-    public OffsetDateTime getTimestamp() {
-        return timestamp;
+    public Optional<ExtraData> getExtraData() {
+        return userContent.getExtraData();
     }
 
-    public ExtraData getExtraData() {
-        return extraData;
+    public ApplicationContent getApplicationContent() {
+        return applicationContent;
     }
 
-    public ExtraData getOtherPartyExtraData() {
-        return otherPartyExtraData;
+    public UserContent getUserContent() {
+        return userContent;
     }
```

Comment:
@laurent-yoti @dhivya-pattabhiraman I made the change we discussed on Fri. This is the solution, among the suggested, I like the most. @jlmartyres is also looking at it.

Reply:
Hum, I am a bit confused...

`getProfile()` : it is acknowledged that the RP using the SDK is interested mostly in that bit, which is the user profile, what you suggesting here is in line with the first proposal, I am only confused by the optional aspect of it... although, in theory, you are right, it could be empty if the user has only shared their `rememberMeId`.

`getExtraData()`: I believe that the outcome of our discussion and from what you saw in the code, these extra data is actually corresponding to the application content's extra data... so it should read `returns optional applicationExtraData`

`getApplicationContent()` --> happy to have that

`getUserContent()` --> happy to have that


@dhivya-pattabhiraman @jlmartyres @irotech ?

Reply:
About the profile (`userProfile`), the SDK needs to account for the missing (null) values and the only field we can guarantee is never null is the `applicationProfile`. As you said a share with a `rememberMeId` only will return a null `userProfile`.

The `getExtraData()` bit is something I didn't quite get. Currently the SDK, in the dynamic scenario, returns the applicationExtradata only completely ignoring the userExtradata. So I asked myself shouldn't it be the extraData Core sets, in share completion, along with the userProfile? Means both the values in the user content?

Reply:
Different taste/strategy for the profile, you could always have a profile with 0 attributes inside, so it would always exist (just a suggestion).

From what we saw in the current implementation of the receipt, there is only 1 `extraData` and it seemed to be linked with the application content, in the sense that it reads `extra_data_content` ... and we kinda concluded  that:
- `profile_content` => application/RP
- `extra_data_content`  => application/RP
- `other_party_profile_content`  => user
- `other_party_extra_data_content`  => user

We can give it a try in our test environment.

Reply:
@pgreci can you please shed a light on these extraData please? We said both, application and user, have a meaning but there is no real use cases (yet). 
The Java SDK, dynamic scenario, returns both profiles but only the applicationExtraData which in Core language is translated in 
```
    optional string other_party_profile_uri = 2;
    optional string profile_uri = 3;
    optional string extra_data_uri = 5;
```

In v2 we want to create an API for the integrator to have a short 
`getProfile()` --> **otherPartyProfile** value
`getExtraData()` --> ??? shouldn't this be **otherPartyExtraData**? Or **extraData** as it is now?

and additional
`getApplicationContent()` (contains both **profile** and **extraData**)
`getUserContent()` (contains both **otherPartyProfile** and **otherPartyExtraData**)

Reply:
@irotech I thought we saw that Connect mapping for the current v1 is returning only the `extra_data` and that this is mapped to the application content... as in, these are the additional data provided by the application (RP) to dictate some of the share operations (location for example).
I think we can quickly check that out on stg1, using both v1 and v2, and see what the extra_data of v1 matches with, user's or application's ones.

Reply:
I did a test with this request

```
        Extension<LocationConstraintContent> locationExtension = new LocationConstraintExtensionBuilder()
                .withLatitude(51.5074)
                .withLongitude(-0.1278)
                .withRadius(6000)
                .build();

        Policy policy = Policy.builder()
                .withWantedRememberMe(true)
                .build();

        ShareSessionRequest shareSessionRequest = ShareSessionRequest.builder()
                .withPolicy(policy)
                .withRedirectUri(new URI("https://host/redirect/"))
                .withExtension(locationExtension)
                .build();
```

for reference the receipt ID; `cy7f1+00qvd1pKSBCzcbzVDCEl2w/jg4evu0Sa9fCbTkMKwVkclE1XijZlNWDVnv`

the wrapped receipt 
```
 id = "cy7f1+00qvd1pKSBCzcbzVDCEl2w/jg4evu0Sa9fCbTkMKwVkclE1XijZlNWDVnv"
 sessionId = "ss.v2.ChY2OHNvQ2RUU1RwT19kWU85dDVlZmhBEgdsZDUuZ2Jy"
 timestamp = {OffsetDateTime@9160} "2023-03-13T16:07:07Z"
 content = {WrappedReceipt$Content@9161} 
  profile = {byte[245]@9168} [10, ... -108, +145 more]
  extraData = null
 otherPartyContent = {WrappedReceipt$Content@9162} 
  profile = null
  extraData = {byte[68]@9169} [10, ... 42, -90] <-- otherPartyContent.extraData has the location value
 wrappedItemKeyId = "218bd625-cb7e-4f3a-a5aa-12c1fde2101f"
 rememberMeId = {byte[48]@9164} [97, ... 113, -77]
 parentRememberMeId = {byte[48]@9165} [-32, ... -92, -89]
 wrappedKey = {byte[48]@9166} [-54, ... 51, -66]
 error = null
```

there is no converter if not for THIRD_PARTY_ATTRIBUTE in the Java SDK so the extraData value for LOCATION is lost

The final receipt came out, as I expected, this way:

```
 id = "cy7f1+00qvd1pKSBCzcbzVDCEl2w/jg4evu0Sa9fCbTkMKwVkclE1XijZlNWDVnv"
 sessionId = "ss.v2.ChY2OHNvQ2RUU1RwT19kWU85dDVlZmhBEgdsZDUuZ2Jy"
 rememberMeId = "YZVN6sW3pbh/f91KXerUk2OrLOhZMFk/nQCBzTwFIa80sb5tAJtTZn/KC72+nXGz"
 parentRememberMeId = "4PpmWo0toxWo5qvnz/kDu7g81LNdo5j2LttLQpffFEnUr/u+cmjfNA2ztznm/qSn"
 timestamp = {OffsetDateTime@9205} "2023-03-13T16:07:07Z"
 applicationContent = {Receipt$ApplicationContent@9230} 
  profile = {ApplicationProfile@9234} 
   protectedAttributes = {Collections$UnmodifiableMap@9235}  size = 4
    "application_logo" -> {ArrayList@9243}  size = 1
    "application_name" -> {ArrayList@9245}  size = 1
    "application_url" -> {ArrayList@9247}  size = 1
    "application_receipt_bgcolor" -> {ArrayList@9249}  size = 1
  extraData = null
 userContent = {Receipt$UserContent@9231} 
  profile = null
  extraData = {ExtraData@9233}  <-- the otherPartyExtraData (user)
   attributeIssuanceDetails = null
 error = null
```